bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/workbench/DynamicMenuItem.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2023 ETAS GmbH and others, all rights reserved.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ETAS GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.ui.workbench;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+
+public class DynamicMenuItem {
```

Comment:
Also, DynamicMenuItem is not a great name, especially if it's meant to serve only the newWizard case. I think it would be better to have this nested in the NewWizardNewPage or something, with more narrowly-scope names for preferences.
So it could be invoked with something like `NewWizardNewPage.getItemsFromPreferences()` or something of that sort that reads more efficient.

Reply:
This cannot be nested in NewWizardNewPage class as it is having default / Package visibility . Can I use NewWizard class instead ? (Need to access from WorkbenchPlugin.java )